src/Npgsql/ConnectorPool.cs

```diff
@@ -468,6 +470,14 @@ void EnablePruning()
         {
             lock (_pruningTimer)
             {
+                if (_concurrentPruningCounter != 0 || _pruningTimerEnabled)
```

Comment:
Since changes happens when the lock is acquired the read will happen only after it. It's not specified in the docs, but `Monitor` enforces a barrier. No need in `volatile`, but feel free to keep it as is for clarity.